src/AdminConsole/Billing/BackgroundServices/MeteredBillingBackgroundService.cs

```diff
@@ -17,8 +21,71 @@ protected override async Task DoWorkAsync(CancellationToken cancellationToken)
         try
         {
             using IServiceScope scope = services.CreateScope();
-            var billingService = scope.ServiceProvider.GetRequiredService<ISharedBillingService>();
-            await billingService.UpdateUsageAsync();
+            var db = scope.ServiceProvider.GetRequiredService<ConsoleDbContext>();
+            var timeProvider = scope.ServiceProvider.GetRequiredService<TimeProvider>();
+
+            // 1. Retrieve all paid organizations.
+            var sharedBillingService = scope.ServiceProvider.GetRequiredService<ISharedBillingService>();
+            var organizations = await sharedBillingService.GetPayingOrganizationsAsync();
+
+            // 2. Populate how many unique users per application we have.
+            Logger.LogInformation("Updating metered billing for {Organizations} organizations.", organizations.Count);
+            foreach (var organization in organizations)
+            {
+                var items = new Dictionary<string, int>();
+
+                foreach (var application in organization.Applications)
+                {
+                    Logger.LogInformation("Processing metered billing for application '{Application}'.", application.Id);
+
+                    var context = scope.ServiceProvider.GetRequiredService<ICurrentContext>();
+                    context.SetApp(application);
+
+                    var request = new PeriodicCredentialReportRequest(DateOnly.FromDateTime(timeProvider.GetUtcNow().Date.AddDays(-1)), null);
+                    var api = scope.ServiceProvider.GetRequiredService<IScopedPasswordlessClient>();
```

Comment:
We initialize the `IScopedPasswordlessClient` implementation here which will use the correct `ApiSecret` to retrieve the metered billing information.